backends/dnf/pk-backend-dnf.c

```diff
@@ -823,6 +823,26 @@ dnf_utils_run_query_with_newest_filter (DnfSack *sack, HyQuery query)
 	return results;
 }
 
+static gboolean
+dnf_utils_force_distupgrade_on_upgrade (DnfSack *sack)
+{
+	HyQuery query_tmp = hy_query_create (sack);
+	hy_query_filter (query_tmp, HY_PKG_REPONAME, HY_EQ, HY_SYSTEM_REPO_NAME);
+	const gchar* distroverpkg_names[] = { "system-release", "distribution-release", NULL };
+	hy_query_filter_provides_in (query_tmp, (gchar**) distroverpkg_names);
+	const gchar* distupgrade_provides[] = { "system-upgrade(dsync)", "product-upgrade() = dup", NULL };
+	hy_query_filter_provides_in (query_tmp, (gchar**) distupgrade_provides);
+	GPtrArray *plist = hy_query_run (query_tmp);
+	gint candidates = plist->len;
+	g_ptr_array_unref (plist);
+	hy_query_free (query_tmp);
```

Comment:
can we use g_autoptr() here?

Reply:
Does that work with non-GObject data types?

Reply:
Well, `plist` is :)

Reply:
Done!